src/connector/src/parser/unified/debezium.rs

```diff
@@ -150,11 +186,38 @@ where
             }
 
             // value should not be None.
-            ChangeEventOperation::Upsert => self
-                .value_accessor
-                .as_ref()
-                .unwrap()
-                .access(&[AFTER, &desc.name], Some(&desc.data_type)),
+            ChangeEventOperation::Upsert => {
+                // For upsert operation, if desc is an additional column, access field in the `SOURCE` field.
+                desc.additional_column.column_type.as_ref().map_or_else(
+                    || {
+                        self.value_accessor
+                            .as_ref()
+                            .expect("value_accessor must be provided for upsert operation")
+                            .access(&[AFTER, &desc.name], Some(&desc.data_type))
+                    },
+                    |additional_column_type| {
+                        match additional_column_type {
+                            &ColumnType::Timestamp(_) => {
+                                tracing::info!(
+                                    "access timestamp field: `{}, type: {:?}, column_type: {:?}`",
+                                    desc.name,
+                                    desc.data_type,
+                                    desc.additional_column.column_type
+                                );
+
+                                // access payload.source.ts_ms
+                                self.value_accessor
+                                    .as_ref()
+                                    .expect("value_accessor must be provided for upsert operation")
+                                    .access(&[SOURCE, SOURCE_TS_MS], Some(&desc.data_type))
```

Comment:
Not ideal but at least better than current status:

Access `DataType::Int64` and call `Timestamptz::from_millis`. Otherwise, accessing a number as timestamptz may interpret it as micros or seconds.